changelog/allow_ambiguous_block_association_to_match_source.md

```diff
@@ -0,0 +1 @@
+* [#10969](https://github.com/rubocop/rubocop/issues/10969): Allow `Lint/AmbiguousBlockAssociation` `AllowedPattern` to match on `send_node.source` instead of `method_name`
```

Comment:
Please refer to the following for file names:
https://github.com/rubocop/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format

Reply:
It may be a little more user-friendly to write it like this:

```suggestion
* [#10969](https://github.com/rubocop/rubocop/issues/10969): Fix a false negative for `AllowedPatterns` of `Lint/AmbiguousBlockAssociation` when using a method chain. ([jcalvert][])
```

Also, prefix the file name with `fix_`:

`changelog/fix_allow_ambiguous_block_association_to_match_source.md`